bindings/matrix-sdk-ffi/src/widget.rs

```diff
@@ -174,6 +174,22 @@ pub struct VirtualElementCallWidgetOptions {
 
     /// Can be used to pass a PostHog id to element call.
     pub analytics_id: Option<String>,
+
+    /// Whether to use e2ee.
+    /// If `per_participant_e2ee` is enabled,
+    /// this will use matrix to exchange keys.
+    /// Otherwise a password in the url is expected.
+    /// (A password should only be used for sharable links and not in embedded
+    /// mode. Passwords are planned to get deprecated and are not supported
+    /// in the rust sdk.)
+    ///
+    /// Default: `true`
+    pub enable_e2ee: Option<bool>,
```

Comment:
Do I understand correctly that the value of `per_participant_e2ee` is only meaningful when `enable_e2ee` is set? If so, it would make more sense to use an `enum` here, perhaps with a name that's slightly more descriptive:

```rust
enum EncryptedKind { // this name is bad but can't think of anything better right now
  NotEncrypted,
  WithPassword,
  WithMatrixKeys, // or something better, since this isn't clear either
}
```

and then the URL can pass a single value that tells how it's encrypted, instead of holding bool values. What do you think?

Reply:
This is definitly more elegant.
I was hoping that the parameters in the rust sdk and the actual url parameters in element call are as similar as possible.
To make it look as familiar as possible to ppl used to element call. (And for the communication with element call devs. This could lead to confusions if an app dev is clainimg he is setting the url to `WithMatrixKeys` which is not a thing in element call.)

I like your proposed api better though. It precisely represents the actual underlying logic. (still I am in favor it kepping it in sync with the element call url params.)

Reply:
Are these booleans part of a MSC, or documented publicaly somewhere?

Reply:
yes, they are here: https://github.com/vector-im/element-call/blob/livekit/docs/url-params.md

Reply:
Ok thanks.

So I've realized that the encoding of the URL happens during serialization of the `ElementCallParams` struct. How about the following:

- we introduce the nicer `enum` at least in the SDK crate, so that's the public interface.
- the FFI layer can keep on using bools, as far as I'm concerned the FFI layer really is for EX anyways.
- when converting from the `VirtualElementCallWidgetOptions` to the `ElementCallParams`, we can expand the enum into the two bools
- then in doc comments above the enum variants, we indicate "this variant corresponds to e2eeEnabled and perParticipantE2ee" etc.

How does that sound?